src/binary/component.rs

```diff
@@ -125,4 +140,29 @@ impl<'a> Component<'a> {
 
         lo.unwrap_or_default()..hi.unwrap_or_default()
     }
+
+    /// Read a note from the note section
+    #[cfg(feature = "backend-sgx")]
+    pub unsafe fn read_note<T: Copy>(&self, name: &str, kind: u32) -> Result<Option<T>> {
+        use std::mem::{size_of, MaybeUninit};
+        use std::slice::from_raw_parts_mut;
+
+        let headers = match self.elf.iter_note_headers(self.bytes) {
+            Some(headers) => headers,
+            None => return Ok(None),
+        };
+
+        for note in headers {
+            let note = note?;
+            if note.name == name && note.n_type == kind && note.desc.len() == size_of::<T>() {
+                let mut value = MaybeUninit::uninit();
+                let ptr = value.as_mut_ptr() as *mut u8;
+                let slice = from_raw_parts_mut(ptr, size_of::<T>());
+                slice.copy_from_slice(note.desc);
```

Comment:
Hmm... Is `note.desc` aligned or do we need `value.as_mut_ptr().read_unaligned(note.desc.as_ptr())` ?

Reply:
I *think* they are equivalent. But I might like your approach better.

Reply:
Fixed.